lib/saml_idp/response_builder.rb

```diff
@@ -39,6 +56,7 @@ def build
         InResponseTo: saml_request_id,
         "xmlns:samlp" => Saml::XML::Namespaces::PROTOCOL do |response|
           response.Issuer issuer_uri, xmlns: Saml::XML::Namespaces::ASSERTION
+          sign response
```

Comment:
Is this an additive change or a destructive change? Like, do we want to have this `sign response` only happen conditionally?

Reply:
This is a great example of just how complicated this code is. `sign response` adds a signature to the response _if_ `sign?` is true, which it only is if called within a `with_signature` block. The conditional is in the controller -- if the signature is requested, it calls `response.signed` vs `response.build`. Both eventually end up calling this `build` method. This line has to be here to ensure the order of the elements in the resulting xml.

Reply:
Yikes, thanks for explaining